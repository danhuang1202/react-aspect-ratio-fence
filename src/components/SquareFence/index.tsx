import { createElement, FunctionComponent, ReactNode } from 'react'
import cns from 'classnames'

// @ts-ignore
import styles from './style.css'

type Props = {
  /**
   *  html tag name
   *  @default div
   * */
  elementType?: string
  /** Custom class name */
  customClass?: string
  /** custom children node */
  children?: ReactNode
}

const SquareFence: FunctionComponent<Props> = ({
  elementType = 'div',
  customClass,
  children
}) =>
  createElement(
    elementType,
    {
      className: cns(styles.wrap, customClass)
    },
    children
  )

export default SquareFence
