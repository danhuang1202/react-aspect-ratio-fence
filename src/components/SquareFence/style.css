.wrap {
  position: relative;
  width: 100%;
}

.wrap:before {
  content: '';
  display: block;
  width: 100%;
  height: 0;
  padding-bottom: 100%;
}

.wrap > :first-of-type {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
