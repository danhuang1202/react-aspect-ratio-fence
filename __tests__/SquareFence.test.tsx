import React from 'react'
import renderer from 'react-test-renderer'
import SquareFence from '@components/SquareFence'

describe('[snapshot] SquareFence', () => {
  it('SquareFence', () => {
    const props = {
      ratio: 16 / 9
    }

    const dom = renderer.create(
      <SquareFence {...props}>
        <img alt="SquareFence" src="SquareFence.png" />
      </SquareFence>
    )
    expect(dom).toMatchSnapshot()
  })

  it('SquareFence with custom elementType', () => {
    const props = {
      elementType: 'span',
      ratio: 16 / 9
    }

    const dom = renderer.create(
      <SquareFence {...props}>
        <img alt="SquareFence" src="SquareFence.png" />
      </SquareFence>
    )
    expect(dom).toMatchSnapshot()
  })
})
