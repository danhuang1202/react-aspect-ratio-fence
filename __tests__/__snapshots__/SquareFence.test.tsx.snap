// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] SquareFence SquareFence 1`] = `
<div
  className="SquareFence_wrap_3jTo2"
>
  <img
    alt="SquareFence"
    src="SquareFence.png"
  />
</div>
`;

exports[`[snapshot] SquareFence SquareFence with custom elementType 1`] = `
<span
  className="SquareFence_wrap_3jTo2"
>
  <img
    alt="SquareFence"
    src="SquareFence.png"
  />
</span>
`;
